cmd/inotify_exporter/main.go

```diff
@@ -0,0 +1,503 @@
+// Copyright 2017 inotify-exporter Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// inotify_exporter monitors file creation, deletion, and other events within
+// an M-Lab experiment data directory. Aggregate event counts are exported for
+// Prometheus monitoring. These events may be logged for offline auditing.
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"path"
+	"regexp"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/rjeczalik/notify"
+	"golang.org/x/sys/unix"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+var (
+	// TODO: support multiple paths.
+	rootPath  = flag.String("path", "", "Path of root directory to watch.")
+	logEvents = flag.Bool("write-logs", false, "Write event logs to YYYY/MM/DD directories.")
+
+	// Patterns for matching date strings in directory names.
+	yearPattern  = regexp.MustCompile("20[0-9]{2}")
+	monthPattern = regexp.MustCompile("0[1-9]|1[0-2]")
+	dayPattern   = regexp.MustCompile("0[1-9]|[12][0-9]|3[0-1]")
+
+	// activeWatches prevents watching the same path twice.
+	activeWatches = &watches{make(map[string]chan notify.EventInfo), sync.Mutex{}}
+)
+
+// Prometheus Metrics.
+var (
+	createCount = prometheus.NewCounter(prometheus.CounterOpts{
+		Name: "inotify_file_create_total",
+		Help: "A running count of all files created in watched directories.",
+	})
+	// createExtensions.WithLabelValues("s2c_snaplog").Inc()
+	createExtensions = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "inotify_extension_create_total",
+			Help: "A running count of file extensions created in watched directories.",
+		},
+		[]string{"ext"},
+	)
+	deleteCount = prometheus.NewCounter(prometheus.CounterOpts{
+		Name: "inotify_file_delete_total",
```

Comment:
Is there any value add to this?  Seems like it is redundant with the accounting by extension.